### ISSUE TITLE
Firefly crashes when the configured connector is not available on start up but not when running

### ISSUE THREAD

**Issue by Chengxuan** [hyperledger/firefly#1277]
There are two aspects of this:
1. we should have consistent behaviour between start-up time and running time. And preferably, firefly shouldn't crash due to one dependency throwing an error as other things could be in a usable state
2. better error surfacing for the errors

**Comment by hosie** [comment]
Seen on `bob` namespace on eap dev cluster on Monday 17 April

[connector.log](https://github.com/hyperledger/firefly/files/11258565/connector.log)
[ff-core.log](https://github.com/hyperledger/firefly/files/11258567/ff-core.log)
[ff-core.restart.log](https://github.com/hyperledger/firefly/files/11258568/ff-core.restart.log)

**Comment by nguyer** [comment]
> Firefly crashes when the configured connector is not available on start up but not when running

This is true, and is working as designed. Now, we may want to change that design. The intention was that if FireFly wasn't configured correctly it would not start, forcing someone to intervene and reconfigure it correctly. As opposed to, if it was configured correctly and had been running for some time, but the connector suddenly became unavailable, it should not crash, but simply wait until it comes back. On startup, there is no way to know the difference between an improper config, or a connector that just isn't there temporarily, so we've taken the "fail fast" approach for now.
